lib/ansible/modules/network/iosxr/iosxr_static_routes.py

```diff
@@ -0,0 +1,228 @@
+                      - Refer to vendor documentation for valid values.
+                    type: int
+                  description:
+                    description:
+                      - Specifies the description for this static route.
+                    type: str
+                  vrflabel:
+                    description:
+                      - Specifies the VRF label for this static route.
+                      - Refer to vendor documentation for valid values.
+                    type: int
+                  tag:
+                    description:
+                      - Specifies a numeric tag for this static route.
+                      - Refer to vendor documentation for valid values.
+                    type: int
+                  track:
+                    description:
+                      - Specifies the object to be tracked.
+                      - This enables object tracking for static routes.
+                    type: str
+                  tunnel_id:
+                    description:
+                      - Specifies a tunnel id for the route.
+                      - Refer to vendor documentation for valid values.
+                    type: int
+  state:
+    description:
+      - The state the configuration should be left in.
+    type: str
+    choices:
+    - merged
+    - replaced
+    - overridden
+    - deleted
+    - gathered
+    - rendered
+    - parsed
+    default: merged
+"""
+EXAMPLES = """
+"""
+RETURN = """
+before:
+  description: The configuration prior to the model invocation.
+  returned: always
+  type: list
+  sample: >
+    The configuration returned will always be in the same format
+     of the parameters above.
+after:
+  description: The resulting configuration model invocation.
+  returned: when changed
+  type: list
+  sample: >
+    The configuration returned will always be in the same format
+     of the parameters above.
+commands:
+  description: The set of commands pushed to the remote device.
+  returned: always
+  type: list
+  sample:
+    - router static
+    - vrf dev_site
+    - address-family ipv4 unicast
+    - 192.0.2.48/28 192.0.2.12 FastEthernet0/0/0/1 track ip_sla_10 description dev1
+    - address-family ipv6 unicast
+    - no 2001:db8:1000::/36
+    - 2001:db8:3000::/36 2001:db8:2000:2::2 FastEthernet0/0/0/4 track ip_sla_11 description prod1
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.iosxr.argspec.static_routes.static_routes import (
+    Static_routesArgs,
+)
+from ansible.module_utils.network.iosxr.config.static_routes.static_routes import (
+    Static_routes,
+)
+
+
+def main():
+    """
+    Main entry point for module execution
+
+    :returns: the result form module invocation
+    """
+    required_if = [
+        ("state", "merged", ("config",)),
+        ("state", "replaced", ("config",)),
+        ("state", "overridden", ("config",)),
```

Comment:
can we have entry for parsed state that requires ('running_config')

Reply:
Resolved.